core/core-impl/src/main/java/ai/timefold/solver/core/api/solver/SolverFactory.java

```diff
@@ -115,6 +116,15 @@ static <Solution_> SolverFactory<Solution_> create(SolverConfig solverConfig) {
      *
      * @return never null
      */
-    Solver<Solution_> buildSolver();
+    default Solver<Solution_> buildSolver() {
+        return this.buildSolver(null);
+    }
 
+    /**
+     * As defined by {@link #buildSolver()}.
+     *
+     * @param configOverride sometimes null, includes settings that override the default configuration
+     * @return never null
+     */
+    Solver<Solution_> buildSolver(SolverConfigOverride<Solution_> configOverride);
```

Comment:
I question whether this is the right way. If you have the argument, and you also have an overload which does not need the argument, then this method should not accept `null`. You shouldn't have two ways of doing the same obvious thing.

Reply:
I have made changes to the logic to ensure that the configOverride is now required.